data/datasets.py

```diff
@@ -228,26 +224,222 @@ def __len__(self):
-            batch_size=opts.data.loaders.get("batch_size", 4),
-            shuffle=opts.model.is_train,
-            num_workers=opts.data.loaders.get("num_workers", 8),
+        if depth is True:
+            return DataLoader(
+                SimDepthDataset(
+                    opts,
+                    transform=transforms.Compose(get_transforms(Dict(opts))),
+                    no_check=no_check,
+                ),
+                batch_size=opts.data.loaders.get("batch_size", 4),
+                shuffle=True,
+                num_workers=opts.data.loaders.get("num_workers", 8),
+            )
+        else:
+            return DataLoader(
+                SimDataset(
+                    opts,
+                    transform=transforms.Compose(get_transforms(Dict(opts))),
+                    no_check=no_check,
+                ),
+                batch_size=opts.data.loaders.get("batch_size", 4),
+                shuffle=True,
+                num_workers=opts.data.loaders.get("num_workers", 8),
+            )
+
+
+class RealSimDepthDataset(Dataset):
+    def __init__(self, opts, transform=None, no_check=False):
+
+        isTrain = opts.model.is_train
+        if isTrain:
+            file_list_path = Path(opts.data.files.train)
+            real_file_list_path = Path(opts.data.real_files.train)
+        else:
+            file_list_path = Path(opts.data.files.val)
+            real_file_list_path = Path(opts.data.real_files.val)
+
+        if file_list_path.suffix == ".json":
+            self.samples_paths = self.json_load(file_list_path)
+            self.real_samples_paths = self.json_load(real_file_list_path)
+        elif file_list_path.suffix in {".yaml", ".yml"}:
+            self.samples_paths = self.yaml_load(file_list_path)
+            self.real_samples_paths = self.yaml_load(real_file_list_path)
+        else:
+            raise ValueError("Unknown file list type in {}".format(file_list_path))
+
+        if not no_check:
+            self.check_samples()
+        self.file_list_path = str(file_list_path)
+        self.real_file_list_path = str(real_file_list_path)
+        self.transform = transform
+        self.opts = opts
+
+    def check_samples(self):
+        """Checks that every file listed in samples_paths actually
+        exist on the file-system
+        """
+        for s in self.samples_paths:
+            for k, v in s.items():
+                assert Path(v).exists(), f"{k} {v} does not exist"
+
+        for s in self.real_samples_paths:
+            for k, v in s.items():
+                assert Path(v).exists(), f"{k} {v} does not exist"
+
+    def json_load(self, file_path):
+        with open(file_path, "r") as f:
+            return json.load(f)
+
+    def yaml_load(self, file_path):
+        with open(file_path, "r") as f:
+            return yaml.safe_load(f)
+
+    def __getitem__(self, i):
+        """Return an item in the dataset with fields:
+        {
+            data: transform({),
```

Comment:
typo?

Reply:
yes, probably from the code of the non depth classes